### DISCUSSION TITLE
Custom templates

### DISCUSSION THREAD

**Discussion by Dadinos** [apidoc/apidoc#1299] [Q&A]
There are some custom templates available on the web for older version of apidoc.
Since the 0.5x version from apidoc they don't work anymore.

I tried to convert them but no luck. Mainly, my bad, because I don't understand how this works.
Is there a discription to make how to setup your own template?
What are the specs for this and how to include own files and styles in the current apidoc version.

Any tips on this?

**Comment by NicolasCARPi** [comment]
Hello,

Sorry no description or spec except looking at the current one and teaking it to your liking.

**Comment by Dadinos** [comment]
Ok additional help question then

I can fix it now when I adapt the writer.js file from the ..\AppData\Roaming\npm\node_modules\apidoc\lib directory.
Is there a apidoc param so I can use my own writer.js instead of the default?

**Reply by NicolasCARPi** [reply]
What you want is to use the template option (`-t`) to feed it a different template, not modify the code.

**Reply by Dadinos** [reply]
Hi Nicolas I understand that and already use:

**apidoc -i ./example -o ./tmp/doc -t d:/apidoc/template -d**

But if a new template has more files and directories then the default template how can I copy these files then to the output directory.
In the writer.js only files are copied from the default template. So I need to create my own copy machine to get them to the output folder. right?

e.q.: from the writer .js (C:\Users\username\AppData\Roaming\npm\node_modules\apidoc\lib)

  ```
createOutputFiles () {
    this.createDir(this.opt.dest);

    // create index.html
    this.log.verbose('Copying template index.html to: ' + this.opt.dest);
    this.fs.writeFileSync(this.path.join(this.opt.dest, 'index.html'), this.getIndexContent());

    // create assets folder
    const assetsPath = this.path.resolve(this.path.join(this.opt.dest + 'assets'));

    const vendorPath = this.path.resolve(this.path.join(this.opt.dest + 'vendor')); <-added
    const jsPath = this.path.resolve(this.path.join(this.opt.dest + 'js'));
    const lokalesPath = this.path.resolve(this.path.join(this.opt.dest + 'locales'));
    const utilsPath = this.path.resolve(this.path.join(this.opt.dest + 'utils'));
    
    this.createDir(assetsPath);
    this.createDir(vendorPath); <- added
    this.createDir(jsPath); <- added
    this.createDir(lokalesPath); <- added
    this.createDir(utilsPath); <- added
    
    // add the fonts
    this.log.verbose('Copying fonts to: ' + assetsPath);
    this.fs.copySync(this.path.join(this.opt.template, 'fonts'), assetsPath);

    // utils added
    this.log.verbose('Copying locales map  to: ' + utilsPath);
    this.fs.copySync(this.path.join(this.opt.template, 'utils'), utilsPath);
    
    // locales added
    this.log.verbose('Copying locales map  to: ' + lokalesPath);
    this.fs.copySync(this.path.join(this.opt.template, 'locales'), lokalesPath);

  etc...
```
So I was wondering If we don't want to modify the default code is there a way to specifiy a own writer path.

Like:

apidoc -i ./example -o ./tmp/doc -t d:/apidoc/template -d **-w d:/apidoc/template/writer.js**

Or is there a better way to do this?

Regards,

Aren

**Reply by NicolasCARPi** [reply]
Nope.
